Shopify/tophat
# Add a download failure message, rather than trying to unzip an HTML page

### What does this change accomplish?

If you try to install a build where the URL it downloads gives you an HTTP 403 response, previously Tophat would ignore the status code and save the resulting response body to the target file. Later it would get a failure trying to unzip this file.

### How have you achieved it?

Tophat now looks for a 200 response status code, and if it is not, it reports it as an error. The failure reason is attached to the the error enum so that we can display it to the user.

### How can the change be tested?

Open up an install URL to a non-existent AWS S3 bucket:

http://localhost:29070/install/ios?virtual=https://some-bucket-we-dont-have-access-to.s3.us-east-1.amazonaws.com/foo/build.app.zip

See the error like this:

<img width="372" alt="Screenshot 2024-10-08 at 11 25 50 AM" src="https://github.com/user-attachments/assets/8e0774a3-2d37-4bdb-a5f4-fee66f5cb27f">
